src/main/query_profiler.cpp

```diff
@@ -87,6 +87,19 @@ QueryProfiler &QueryProfiler::Get(ClientContext &context) {
 	return *ClientData::Get(context).profiler;
 }
 
+void QueryProfiler::Reset(const bool running_p, string query) {
```

Comment:
Can we split this into two separate functions? Maybe `Start(string query)` and `Reset()`? `Start` can then call `Reset()`.